pyaedt/modules/MonitorIcepak.py

```diff
@@ -496,75 +507,95 @@ def _delete_removed_monitors(self):
                         i + " monitor object lost its assignment due to geometry modifications and has been deleted."
                     )
 
-
-class PointMonitor(Monitor):
-    def __init__(self, monitor_name, monitor_type, point_id, quantity, app):
-        self._name = monitor_name
-        self._type = monitor_type
-        self._id = point_id
-        self._quantities = quantity
-        self._app = app
-
-    @property
-    def name(self):
+    @pyaedt_function_handler()
+    def insert_monitor_object_from_dict(self, monitor_dict, mode=0):
         """
-        Get the name of the monitor point
+        Get the object that the monitor is applied to
+
+        Parameters
+        ----------
+        monitor_dict : dict
+           Dictionary containing monitor object information
+        mode : int
+            Integer to select the information to handle. To identify faces, vertices, surfaces and object to which the
+            monitor is to be assigned it is possible to use:
+                ids and names, mode=0, required dict keys: "Name", "Type", "ID", "Quantity"
+                positions, mode=1, required dict keys: "Name", "Type", "Geometry Assignment", "Location", "Quantity"
 
         Returns
         -------
-        str
-        """
-        return self._name
+            Name of the monitor object.
+        """
+        m_case = monitor_dict["Type"]
+        m_quantity = monitor_dict["Quantity"]
+        m_name = monitor_dict["Name"]
+        m_object = None
+        if mode == 1:
+            if m_case == "Face":
+                for f in self._app.modeler.get_object_from_name(monitor_dict["Geometry Assignment"]).faces:
+                    if f.center == monitor_dict["Location"]:
+                        m_object = f.id
+                        break
+            elif m_case == "Vertex":
+                for v in self._app.modeler.get_object_from_name(monitor_dict["Geometry Assignment"]).vertices:
+                    if v.position == monitor_dict["Location"]:
+                        m_object = v.id
+                        break
+            elif m_case == "Object":
+                m_object = monitor_dict["Geometry Assignment"]
+            elif m_case == "Surface":
+                m_object = monitor_dict["Geometry Assignment"]
+        elif mode == 0:
+            m_object = monitor_dict["ID"]
+        else:
+            self._app.logger.error("Only modes supported are 0 and 1")
+        if m_object is None:
+            self._app.logger.error("{} monitor object could not be restored".format(m_name))
+            return False
+        self._app.configurations.update_monitor(m_case, m_object, m_quantity, m_name)
+        return m_name
 
-    @property
-    def id(self):
-        """
-        Get the point name, object name or vertex id as appropriate
 
-        Returns
-        -------
-        str or int
-        """
-        return self._id
+class ObjectMonitor:
```

Comment:
add a short description to the class
```suggestion
class ObjectMonitor:
    """Provides Icepak Monitor methods and properties."""
```